### ISSUE TITLE
fix(setTranslation): change not triggered

### ISSUE THREAD

**Issue by ali-idrizi** [ngx-translate/core#788]
Is there currently any issue with setTranslation or am I doing something wrong?

My translation for `en` in the i18n directory looks like this:
```
{
  "home": {
    "title": "Title"
  }
}
```

I am running the following code in AppComponent

```
this.languageService.setTranslation('en', {test: 'test'}, true);
this.languageService.getTranslation('en').subscribe(res => console.log(res));
```

And what I get in the console is nothing different than the initial JSON file.

I am using Angular v5.2.3, ngx-translate v9.1.1 with ngx-translate-messageformat-compiler

**Comment by martinmicunda** [comment]
I have the same issue.

**Comment by jlberrocal** [comment]
that's because order, if you emit the change before subscribing you might not see the change, but if you change the calling order you should be able to see the new properties

**Comment by ali-idrizi** [comment]
Nope, still nothing. It doesn't even call the `.next()` after `setTranslation`, nothing shows in the console. Here is a [Plunker](http://plnkr.co/edit/UVVFgLmcbF8VZF2AViJX) demo

**Comment by jlberrocal** [comment]
ok, seems that @ocombe set the `getTranslation`  method to be a one time subscription, so you will no be able to see the changes, if you want to know when a change take place you will have to listen for `onTranslationChange` event instead, however @ocombe there is an issue because translate pipe is never seeing the change please refer to this [plunk](http://plnkr.co/edit/zE3vW1FJrDgnx99txjoB?p=preview)

**Comment by gustavshf** [comment]
I'm having the exact same issue, spent a day debugging translation issues, turns out this was the problem. Any plans on fixing this?

**Comment by nguyenbathanh** [comment]
It looks like doesn't work with translation pipe as @jlberrocal mentioned.

**Comment by DovakhinMRK** [comment]
Any updates on this?

**Comment by qwekelly** [comment]
Any updates on this?

**Comment by rbalet** [comment]
After a digging a little bit, I found that, if you do the following 

```typescript
setTimeout(() => {
  this.languageService.setTranslation('en', {test: 'test'}, true);
}, 100)
```

Then the test will be added to the list, but with a timeout of `0` this doesn't work.  
It is indeed logical that this happens as, in the actual code, the library has to fetch first the file. 

After fetching it then set the translation, but then ignore any changes that had happened before. 

I'll do a PR that will make the library aware of changes that happens **before** the i18n files have been loaded

**Comment by rbalet** [comment]
Closing, as the v18 should fix this issue
